plugin/storage/integration/badgerstore_test.go

```diff
@@ -30,22 +28,21 @@ import (
 
 type BadgerIntegrationStorage struct {
 	StorageIntegration
-	logger *zap.Logger
+	logger  *zap.Logger
+	factory *badger.Factory
 }
 
 func (s *BadgerIntegrationStorage) initialize() error {
-	f := badger.NewFactory()
-
-	err := f.Initialize(metrics.NullFactory, zap.NewNop())
+	err := s.factory.Initialize(metrics.NullFactory, zap.NewNop())
```

Comment:
This doesn't look quite right, since `initialize()` function is called multiple times, but I am not sure that `factory.Initialize()` is meant to be called more than once. I would suggest moving it into `TestBadgerStorage()` if you want to reuse the factory.

And another issue is that whether `cleanUp()` function is supposed to reset the storage to empty. If so, I would create new factory in `initialize()` every time and close it in `cleanUp()`.

Reply:
^

Reply:
@yurishkuro I have updated the PR. May I ask if this is what was intended.